### PR TITLE
[Excel] (ribbon) open task panes from contextual tabs

### PR DESCRIPTION
**Pull request by Rick-Kirkham**

### CODE REVIEW

File: docs/design/contextual-tabs.md [hunk 1]
```diff
@@ -373,6 +373,91 @@ function myContextChanges() {
...
+                // some control properties omitted
+            }          ]
+        }
+      ]
+    }
+  ]
+}`
+```
+
 ## Localizing the JSON blob
```
Comment (lindalu-MSFT) [inline docs/design/contextual-tabs.md:461]:
```suggestion
## Localize the JSON blob
```
I checked the article, and this is the only heading that uses a gerund verb. Suggest changing to align with other headings in the topic.

**Review by lindalu-MSFT — CHANGES_REQUESTED** [review]
I didn't address the "you cans" in this topic, but there are 10 instances. Suggest you reduce the number of times you use this phrase. For example, in the introductory paragraph:

"You can include custom contextual tabs in your Office Add-in and specify when they are visible or hidden, by creating event handlers that change the visibility."

can be:

"Include custom contextual tabs in your Office Add-in and specify when they are visible or hidden, by creating event handlers that change the visibility."

**Comment by Rick-Kirkham** [comment]
> I didn't address the "you cans" in this topic, but there are 10 instances. Suggest you reduce the number of times you use this phrase. For example, in the introductory paragraph:
> 
> "You can include custom contextual tabs in your Office Add-in and specify when they are visible or hidden, by creating event handlers that change the visibility."
> 
> can be:
> 
> "Include custom contextual tabs in your Office Add-in and specify when they are visible or hidden, by creating event handlers that change the visibility."

I've fixed this in places where it can be. In some places, the "you can" is appropriate because it expresses an option, not something the reader must do.

**Comment by lindalu-MSFT** [comment]
@Rick-Kirkham That's a great way to differentiate. If it's an option, then "you can" is fine. If not, try to write the sentence differently to avoid "you can."

File: docs/design/contextual-tabs.md [hunk 2]
```diff
@@ -373,7 +373,92 @@ function myContextChanges() {
...
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}`
+```
+
+In a task pane, to open a page that is at a different URL from the URL of the shared runtime, specify a `sourceLocation` property in the definition of the action. In the following example, a second task pane is opened from a different button.
```
Comment (davidchesnut) [inline docs/design/contextual-tabs.md:411]:
```suggestion
To open any task pane that is not the default task pane, specify a `sourceLocation` property in the definition of the action. In the following example, a second task pane is opened from a different button.
```

File: docs/design/contextual-tabs.md [hunk 3]
```diff
@@ -373,7 +373,92 @@ function myContextChanges() {
...
+                "label": "Open Tables Taskpane",
+                // some control properties omitted
+            }          ]
+        }
+      ]
+    }
+  ]
+}`
+```
+
```
Comment (davidchesnut) [inline docs/design/contextual-tabs.md:460]: You should also document the `title` and `supportPinning` properties for opening a task pane.
```json
  {
            "id": "showTaskpaneContoso",
            "type": "ShowTaskpane",
            "title": "Contoso task pane",
            "supportPinning": false
          }
```

File: docs/design/contextual-tabs.md [hunk 4]
```diff
@@ -373,7 +373,92 @@ function myContextChanges() {
...
+## Localize the JSON blob
```
Comment (davidchesnut) [inline docs/design/contextual-tabs.md:461]: I think this meant to say JSON. Not a binary large object (BLOB).
```suggestion
## Localize the JSON text
```

File: docs/design/contextual-tabs.md [hunk 5]
```diff
@@ -373,7 +373,92 @@ function myContextChanges() {
...
+    }
+  ]
+}`
+```
+
+In a task pane, to open a page that is at a different URL from the URL of the shared runtime, specify a `sourceLocation` property in the definition of the action. In the following example, a second task pane is opened from a different button.
+
+> [!IMPORTANT]
+>
+> - When a `sourceLocation` is specified for the action, then the task pane does *not* use the shared runtime.
```
Comment (davidchesnut) [inline docs/design/contextual-tabs.md:415]:
```suggestion
> - When a `sourceLocation` is specified for the action, then the task pane does *not* use the shared runtime. It runs in a new JavaScript runtime.
```

**Review by davidchesnut — CHANGES_REQUESTED** [review]
Looking good. A few suggestions. Thanks!

**Pull request merged**